### ISSUE TITLE
Rethrowing duplication errors as CodecException

### ISSUE THREAD

**Issue by Warmek** [eclipse/leshan#1376]
### Question

When a SEND containing a duplicates (same path + timestamp) is received, TimestampedLwM2mNodes.Builder::build throws an IllegalArgumentException.
```java
if (noDuplicate && previous != null) {
    throw new IllegalArgumentException(String.format(
             "Unable to create TimestampedLwM2mNodes : duplicate value for path %s. (%s, %s)",
             internalNode.path, internalNode.node, previous));
}

```
That exception is just catched in SendResource::handlePOST, to be sent to the SendListener::onError callback and rethrown.
```java
} catch (RuntimeException e) {
    sendHandler.onError(registration, e);
    throw e;
}
```

Maybe it would be better to catch that exception (in LwM2mNodeSenMLDecoder::decodeTimestampedNodes) and rethrow as CodecException?
```java
} catch (SenMLException | IllegalArgumentException e) {
        String hexValue = content != null ? Hex.encodeHexString(content) : "";
        throw new CodecException(e, "Unable to decode nodes : %s", hexValue, e);
}
```

**Comment by sbernard31** [comment]
Yep totally agree ! Good catch :+1:  !

**Comment by Warmek** [comment]
I've made a commit on [opl/rethrow_illegalArgumentException](https://github.com/eclipse/leshan/compare/master...JaroslawLegierski:leshan:opl/rethrow_illegalArgumentException)

**Comment by sbernard31** [comment]
Yep I think you can create a PR

**Comment by sbernard31** [comment]
Fix in `master` by #1377.

Thx @Warmek for catching this :wink: !
